persistence-jdbc/core/src/main/resources/reference.conf

```diff
@@ -6,12 +6,64 @@ jdbc-defaults.slick {
   # set to one of: slick.jdbc.PostgresProfile$, slick.jdbc.MySQLProfile$, slick.jdbc.OracleProfile$ or slick.jdbc.H2Profile$
   # profile = "slick.jdbc.PostgresProfile$"
 
-  # The JNDI name
-  jndiName = "DefaultDS"
+  # The JNDI name for the Slick pre-configured DB
+  # By default, this value will be used by all akka-persistence-jdbc plugin components (journal, read-journal and snapshot).
+  # you may configure each plugin component to use different DB settings
+  jndiDbName=DefaultDB
```

Comment:
I was wondering if the following would be better:

```
jndiDbNames {
    journal = DefaultDB
    read-journal = DefaultDB
    snapshot = DefaultDB
}
```
the¡n in the overrider...

```
jdbc-journal.slick.jndiName=${jdbc-defaults.slick.jndiDbNames.journal}
jdbc-read-journal.slick.jndiName=${jdbc-defaults.slick.jndiDbNames.read-journal}
jdbc-snapshot-store.slick.jndiName=${jdbc-defaults.slick.jndiDbNames.snapshot}
```

That way we will be more explicity about the fact that we are configuring one default DB and use it on the three components, but that we could define other DB and corresponding DataSource as well.

WDYT?

Reply:
Or maybe even better:

```
jdbc-defaults.slick  { 
  journal.jndiDbName = DefaultDB
  read-journal.jndiDbName = DefaultDB
  snapshot.jndiDbName = DefaultDB
}
```

Reply:
I think the idea with the existing structure is if you want to change them all together (for some reason) there is a single place to do so. Maybe not a common use case, but OTOH I don't see any advantage for users to customizing `jdbc-defaults.slick.journal.jndiDbName` instead of customizing `jdbc-journal.slick.jndiName` directly.

Reply:
I think it's the other way around. We now force the same connection pool and async executor for all components but write and read side have, most often, different requirements. Same for the snapshotting part. 

A thread willing to persist a snapshot is now competing with the journal thread to have a DB connection. 

Maybe we need more time to evaluate it and make a decision. In any case, this should not block this PR. Users can still define different data sources and use them on each plugin component. It's not documented, but it's possible. 

BTW, this PR still need a revised documentation.

Reply:
I don't think we force it... we just have it set that way by default (for convenience). Why do you say force?

Reply:
My wording is not correct and I'm contradicting myself as well. 

We don't force. The option is there, but maybe not obvious.

Reply:
Just now I realised that I totally missed your original point. Yes, indeed, it makes not sense to have defaults for each component. In case of override users should override it directly. 

Sorry for the misunderstanding.